### PR TITLE
refactor: move more code from logical plans to ops

### PR DESCRIPTION
**Pull request by systay**
## Description
We have good support for rewriting our query plans in the operator side of the planner. This PR moves a few constructs that were still planned on the logicalPlan structures over to the operators.

The remaining responsibility of the logicalPlan implementations is now to build the engine primitive tree.

## Related Issue(s)
Tracking issue https://github.com/vitessio/vitess/issues/11626

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/planbuilder/operator_transformers.go [hunk 1]
```diff
@@ -448,36 +446,60 @@ func newRoutingParams(ctx *plancontext.PlanningContext, opCode engine.Opcode) *e
...
+	}
+	directives := cmt.Directives()
+	scatterAsWarns := directives.IsSet(sqlparser.DirectiveScatterErrorsAsWarnings)
+	timeout := queryTimeout(directives)
+	multiShardAutoCommit := directives.IsSet(sqlparser.DirectiveMultiShardAutocommit)
+	return &queryHints{
+		scatterErrorsAsWarnings: scatterAsWarns,
+		multiShardAutocommit:    multiShardAutoCommit,
+		queryTimeout:            timeout,
+	}
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/operator_transformers.go:467]: nit:
```suggestion
	return &queryHints{
		scatterErrorsAsWarnings: directives.IsSet(sqlparser.DirectiveScatterErrorsAsWarnings),
		multiShardAutocommit:    directives.IsSet(sqlparser.DirectiveMultiShardAutocommit),
		queryTimeout:            queryTimeout(directives),
	}
```

File: go/vt/vtgate/planbuilder/operator_transformers.go [hunk 2]
```diff
@@ -734,3 +777,22 @@ func transformLimit(ctx *plancontext.PlanningContext, op *operators.Limit) (logi
...
+
+	if limit.Offset != nil {
+		pv, err = evalengine.Translate(limit.Offset, nil)
+		if err != nil {
+			return nil, vterrors.Wrap(err, "unexpected expression in OFFSET")
+		}
+		plan.elimit.Offset = pv
+	}
+
+	return plan, nil
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/operator_transformers.go:797]: Shouldn't we use `vterrors` errors here?

**Review by frouioui — APPROVED** [review]
Awesome refactoring! No test expectations have changed and this is 💯 Thank you!

File: go/vt/vtgate/planbuilder/operator_transformers.go (hunk 2 above)
Comment (systay) [reply go/vt/vtgate/planbuilder/operator_transformers.go:797]: I just moved this method from `postprocess.go`. It's not wrong to use `Wrap`, right?

**Review by GuptaManan100 — APPROVED** [review]

**Pull request merged**